frontend/src/components/node/Details.tsx

```diff
@@ -150,82 +172,165 @@ export default function NodeDetails() {
     );
   }
 
+  function DrainDialog() {
+    function useDrain(drainNode: Node) {
+      setDrainDialogOpen(false);
+      handleNodeDrain(drainNode);
+    }
+
+    return (
+      <Dialog open={drainDialogOpen} onClose={() => setDrainDialogOpen(false)}>
```

Comment:
Please use the ConfirmDialog we have for making these cases consistent. This avoids e.g. having Cancel / Drain when the options should be No / Yes.

Reply:
Done!